HACKING.rst.txt

```diff
@@ -277,6 +277,12 @@ Therefore if you copied existing code with such a license or made a direct
 translation to Python of it, your code can't be included.  See also `license
 compatibility`_.
 
+A special case is the Numerical Recipes book: it's a book with arguably
+excellent explanations of numerical algorithms, useful as a reference, and that
+also has software that we can't use. Therefore, it is fine to refer to its
+*text* in the documentation, but we cannot accept *code* derived from Numerical
+Recipes.
```

Comment:
The best way to ensure what you write is not based on NR code, is to not
read NR, given that the book contains the code listings. Ditto for
reading other non-license compatible code bases (eg Octave), this is not
specific to NR.
.
IANAL, but to my understanding the question of whether you wrote derived
work of code you read previously is not something you necessarily want
to face, which probably explains the existence of shenanigans such as
https://en.wikipedia.org/wiki/Clean_room_design